### PR TITLE
chore: feature detection for gzip and endpoint override

### PR DESCRIPTION
**Pull request by kuhe**
adds feature detection for:
- custom endpoint
- gzip compression

### CODE REVIEW

**Comment by kuhe** [comment]
- [ ] todo: cyclic dependency

File: packages/core/src/protocols/resolve-path.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { extendedEncodeURIComponent } from "./extended-encode-uri-component";
```
Comment (kuhe) [inline packages/core/src/protocols/resolve-path.ts:1]: this file is not new, copied from another location and not detected by git

File: packages/core/src/protocols/extended-encode-uri-component.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
```
Comment (kuhe) [inline packages/core/src/protocols/extended-encode-uri-component.ts:1]: this file is not new, copied from another location and not detected by git

File: packages/core/src/protocols/collect-stream-body.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { SerdeContext } from "@smithy/types";
```
Comment (kuhe) [inline packages/core/src/protocols/collect-stream-body.ts:1]: this file is not new, copied from another location and not detected by git

**Review by JordonPhillips — APPROVED** [review]

**Review by siddsriv — APPROVED** [review]

**Pull request merged**